src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -221,7 +220,7 @@ impl Validator<'a, 'mir, 'tcx> {
         let Item { tcx, body, def_id, const_kind, ..  } = *self.item;
 
         let use_min_const_fn_checks =
-            tcx.is_min_const_fn(def_id)
+            (const_kind == Some(ConstKind::ConstFn) && tcx.is_min_const_fn(def_id))
```

Comment:
Why does this (have to) change now?

Reply:
Because before I rebased, we only ever called `is_min_const_fn` on functions, but here it was called on constants and statics, too. This wasn't a problem until my changes to `is_min_const_fn`, because until my changes happened `is_min_const_fn` never called `fn_sig` on the given `DefId`, now it does. Basically `is_min_const_fn` now ICEs if you call it on a non-fn